src/components/RegisterWallet/DownloadColdcardConfig.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+import { Button } from "@material-ui/core";
+import { useDispatch, useSelector } from "react-redux";
+import { COLDCARD, ConfigAdapter } from "unchained-wallets";
+import { getWalletConfig } from "../../selectors/wallet";
+import { setErrorNotification } from "../../actions/errorNotificationActions";
+import { downloadFile } from "../../utils";
+
+const DownloadColdardConfigButton = ({ ...otherProps }) => {
+  const [isActive, setIsActive] = useState(false);
+  const walletConfig = useSelector(getWalletConfig);
+  const dispatch = useDispatch();
+
+  const interaction = new ConfigAdapter({
+    KEYSTORE: COLDCARD,
+    jsonConfig: walletConfig,
+  });
+
+  const downloadWalletDetails = () => {
+    setIsActive(true);
+    try {
+      const { startingAddressIndex } = walletConfig;
+      // If this is a config that's been rekeyed, note that in the name.
+      walletConfig.name =
+        startingAddressIndex === 0
+          ? walletConfig.name
+          : `${walletConfig.name}_${startingAddressIndex.toString()}`;
```

Comment:
```typescript 
if (startingAddressIndex === 0) {
   walletConfig.name = `${walletConfig.name}_${startingAddressIndex.toString()}`;
}

```
Might be a bit more straightforward

Reply:
The downside of adding better lint tooling is that a whole bunch of extraneous code gets pulled into the diff! 😅

Reply:
Ohhh that's so annoying. ok ok  🤷🏻

Reply:
Well I'm almost done reviewing this fully, but I guess when that's the case, might be helpful to highlight areas you want review focused on specifically

Reply:
Yeah, totally fair. I usually try and be better about that. Sorry!